src/main/webapp/dev/js/common/instructorFeedbackResults.js

```diff
@@ -151,45 +155,39 @@ function expandOrCollapsePanels(expandCollapseButton, panels) {
     const STRING_COLLAPSE = 'Collapse';
     const expandCollapseStudentsButton = 'a[id^="collapse-panels-button-section-"]';
     const expandCollapseTeamButton = 'a[id^="collapse-panels-button-team-"]';
-    const isButtonInExpandMode = $(expandCollapseButton).html().trim().startsWith(STRING_EXPAND);
     const isCollapsePanelsButton = !panels;
-    /*
-     * When collapsing expandCollapseButton, expandCollapseTeamButton and expandCollapseStudentButton
-     * should be in collapsed state, so that next time when we expand the main panel, we find
-     * the expandCollapseTeamButton and expandCollapseStudentButton in collapsed state.
-     */
-    if (isCollapsePanelsButton && !isButtonInExpandMode) {
-        if ($(expandCollapseStudentsButton).length
-            && $(expandCollapseStudentsButton).html().trim().startsWith(STRING_EXPAND)) {
-            replaceButtonHtmlAndTooltipText(expandCollapseStudentsButton, STRING_EXPAND, STRING_COLLAPSE);
-        }
-        if ($(expandCollapseTeamButton).length
-            && $(expandCollapseTeamButton).html().trim().startsWith(STRING_EXPAND)) {
-            replaceButtonHtmlAndTooltipText(expandCollapseTeamButton, STRING_EXPAND, STRING_COLLAPSE);
-        }
-    }
-    // When collapsing expandCollapseTeamButton, expandCollapseStudentButton should be in collapsed state.
-    if ($(expandCollapseStudentsButton).length && (expandCollapseButton === $(expandCollapseTeamButton))
-        && $(expandCollapseTeamButton).html().trim().startsWith(STRING_COLLAPSE)
-        && $(expandCollapseStudentsButton).html().trim().startsWith(STRING_EXPAND)) {
-        replaceButtonHtmlAndTooltipText(expandCollapseStudentsButton, STRING_EXPAND, STRING_COLLAPSE);
-    }
 
     // {@code panels} is not defined when {@code expandCollapseButton} is collapse panels button. We
     // need to define corresponding {@code targetPanels}.
     const targetPanels = panels || $('div.panel-collapse');
 
-    if (isButtonInExpandMode) {
+    if (isInExpandMode(expandCollapseButton)) {
+        // When expanding all panels via the expand panels button update the state for
+        // all expand/collapse buttons underneath (namely, team and student buttons).
+        if (isCollapsePanelsButton) {
```

Comment:
You shifted the code into the wrong block, right? Your code belongs to the else block, where the button is in collapse mode. I was wondering why my comment was edited from "collapsing" to "expanding".

Reply:
@whipermr5 I shifted the code to if block, so that tooltips and button html get replaced only when we want to expand panels not after collapsing the panels. Doing this ensures that on clicking panel heading, we get expected tooltips.

Reply:
Does this approach work when we use the collapse all button but expand the panels manually by clicking on their headings?

Reply:
No, it doesn't work. I guess I should come up with a better approach to ensure panels headings and buttons are in sync.